hashicorp-vault/main.tf

```diff
@@ -0,0 +1,224 @@
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
+  token                  = data.aws_eks_cluster_auth.cluster.token
+}
+
+resource "kubernetes_namespace" "this" {
+  count = var.chart_create_namespace ? 1 : 0
+  metadata {
+    annotations = {
+      name = var.chart_namespace
+    }
+
+    name = var.chart_namespace
+  }
+}
+
+resource "aws_kms_key" "this" {
+  count                   = var.s3_create_bucket ? 1 : 0
+  description             = "Key for vault s3 bucket"
+  deletion_window_in_days = 10
+}
+
+
+resource "aws_s3_bucket" "this" {
+  count  = var.s3_create_bucket ? 1 : 0
+  bucket = var.s3_bucket_name
+  acl    = "private"
+
+  tags = var.tags
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        kms_master_key_id = aws_kms_key.this.0.arn
+        sse_algorithm     = "aws:kms"
+      }
+    }
+  }
+}
+
+resource "aws_iam_user_policy" "this" {
+  count = var.s3_storage ? 1 : 0
+  name  = "VaultUserPermissions"
+  user  = aws_iam_user.this.0.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "s3:*"
+        ]
+        Effect   = "Allow"
+        Resource = "*"
+      },
+    ]
+  })
+}
+
+resource "aws_iam_policy_attachment" "this" {
+  count      = var.s3_storage ? 1 : 0
+  name       = "test-attachment"
+  users      = [aws_iam_user.this.0.name]
+  roles      = []
+  groups     = []
+  policy_arn = aws_iam_policy.this.0.arn
+}
+
+
+resource "aws_iam_policy" "this" {
+  count       = var.s3_storage ? 1 : 0
+  name_prefix = "vault-kms"
+  description = "EKS VAULT policy for cluster"
+  policy      = data.aws_iam_policy_document.this.0.json
+}
+
+data "aws_iam_policy_document" "this" {
+  count = var.s3_storage ? 1 : 0
+  statement {
+    sid    = "VaultOwn"
+    effect = "Allow"
+
+    actions = [
+      "kms:*"
+    ]
+
+    resources = [aws_kms_key.this.0.arn]
+  }
+}
+
+# TODO test usage of iam role and try to read credentials from secret
+resource "aws_iam_user" "this" {
+  count = var.s3_storage ? 1 : 0
+  name  = "${var.cluster_name}-vault"
+  path  = "/vault/"
+
+  tags = var.tags
+}
+
+resource "aws_iam_access_key" "this" {
+  count = var.s3_storage ? 1 : 0
+  user  = aws_iam_user.this.0.name
+}
+
+resource "local_file" "this" {
+  count    = local.argocd_enabled
+  content  = yamlencode(local.app)
+  filename = "${var.argocd.path}/${local.name}.yaml"
+}
+
+
+resource "helm_release" "this" {
+  count      = 1 - local.argocd_enabled
+  depends_on = [kubernetes_namespace.this]
+  name       = local.name
+  repository = local.repository
+  chart      = local.chart
+  namespace  = local.namespace
+
+  dynamic "set" {
+    for_each = local.conf
+
+    content {
+      name  = set.key
+      value = set.value
+    }
+  }
+}
+
+locals {
+  name           = var.chart_name
+  repository     = "https://helm.releases.hashicorp.com"
+  chart          = "vault"
```

Comment:
You can use var.chart_name here too

Reply:
I think it is not a good idea, cuz name "vault" related to chart name in repo.